tls/s2n_config.c

```diff
@@ -632,7 +632,13 @@ int s2n_config_set_client_hello_cb(struct s2n_config *config, s2n_client_hello_f
 {
     config->client_hello_cb = client_hello_cb;
     config->client_hello_cb_ctx = ctx;
+    config->client_hello_cb_mode = S2N_CLIENT_HELLO_CB_BLOCKING;
+    return 0;
+}
 
+int s2n_config_set_client_hello_cb_mode(struct s2n_config *config, s2n_client_hello_cb_mode cb_mode)
+{
```

Comment:
Nit: `ENSURE_REF`? And maybe a check that `cb_mode` is one of the known modes?

Reply:
done